database/database.go

```diff
@@ -237,3 +251,76 @@ func connectionString(creds DBCredentials) string {
 		mongoWriteConcernTimeout,
 	)
 }
+
+// ensureDBSchema checks that we have all collections and indexes we need and
+// creates them if needed.
+// See https://docs.mongodb.com/manual/indexes/
+// See https://docs.mongodb.com/manual/core/index-unique/
+func ensureDBSchema(ctx context.Context, db *mongo.Database) error {
+	// schema defines a mapping between a collection name and the indexes that
+	// must exist for that collection.
+	schema := map[string][]mongo.IndexModel{
+		dbUsersCollection: {
+			{
+				Keys:    bson.D{{"sub", 1}},
+				Options: options.Index().SetName("usersSubUnique").SetUnique(true),
+			},
+		},
+		dbSkylinksCollection: {
+			{
+				Keys:    bson.D{{"skylink", 1}},
+				Options: options.Index().SetName("skylinksSkylinkUnique").SetUnique(true),
+			},
+		},
+		dbUploadsCollection: {
+			{
+				Keys:    bson.D{{"user_id", 1}},
+				Options: options.Index().SetName("uploadsUserID"),
+			},
+			{
+				Keys:    bson.D{{"skylink_id", 1}},
+				Options: options.Index().SetName("uploadsSkylinkID"),
+			},
+		},
+		dbDownloadsCollection: {
+			{
+				Keys:    bson.D{{"user_id", 1}},
+				Options: options.Index().SetName("downloadsUserID"),
+			},
+			{
+				Keys:    bson.D{{"skylink_id", 1}},
+				Options: options.Index().SetName("downloadsSkylinkID"),
+			},
+		},
+	}
+	for collName, models := range schema {
+		coll, err := ensureCollection(ctx, db, collName)
+		if err != nil {
+			return err
+		}
+		iv := coll.Indexes()
+		names, err := iv.CreateMany(ctx, models)
+		if err != nil {
+			return errors.AddContext(err, "failed to create indexes")
+		}
+		logrus.Debugf("Created new indexes: %v\n", names)
```

Comment:
I'm a bit confused by the usage of `logrus` here, you added it for just this line. Are we going to use it throughout? If we log the successful creation of indices, shouldn't we be logging failed attempts and/or errors as well?

Reply:
I wanted to have some structured logging. `logrus` looked like a decent mature choice. I intend to add logging to many more places, so we can know what's going on.